### ISSUE TITLE
Verify Unet usage

### ISSUE THREAD

**Issue by bspitzbart** [iceberg-project/Rivers#5]
@lmhieu612 Would you have a chance to check how Samira set up Unet?  It works fine, just wondering if anything subtle is missing or if it can be done more efficiently. Thanks!
The relevant code is:
https://github.com/iceberg-project/Rivers/blob/devel_samira/src/training/train_unet.py
https://github.com/iceberg-project/Rivers/blob/devel_samira/src/training/unet_model.py

**Comment by hieulem** [comment]
Hi Samira,
The code looks great to me. Below are some comments I have:
- The training/testing splits should be on the image-level, not patch-level. If you use one patch cropped from a river X for training, the network would work for other image patches cropped from X fairly well - this scenario should be avoided.
- The data augmentation part can be improved. You can apply a sequence of transformation steps rather than choosing only one. You can try to rotate the training images by a small degree and not just 90. 
- I am guessing that Keras takes care of all hyper-parameters (number of epoch, learning rate, beta1, beta2 of Adam, early stopping conditions,...) for you. Playing with those could make big differences. In the long term, you might be considering switching to Pytorch so that you can have complete control over the training.
Other than that, it looks great :). Good job.
